src/dex-ui/hooks/dao/useSendProposalTransaction.tsx

```diff
@@ -0,0 +1,80 @@
+import { useMutation, useQueryClient } from "react-query";
+import { TransactionResponse } from "@hashgraph/sdk";
+import { DAOMutations, DAOQueries } from "./types";
+import { DexService } from "@services";
+import { useDexContext } from "@hooks";
+import { isNil } from "ramda";
+import { MemberOperationsType } from "@dex-ui/pages/dao/MemberUpdate/types";
+
+interface AddMemberForm {
+  newMemberAddress: string;
+  safeAccountId: string;
+  threshold: number;
+  multiSigDAOContractId: string;
+}
+
+interface DeleteMemberForm {
+  memberAddress: string;
+  safeAccountId: string;
+  threshold: number;
+  multiSigDAOContractId: string;
+}
+
+interface ReplaceMemberForm {
+  newMemberAddress: string;
+  oldMemberAddress: string;
+  safeAccountId: string;
+  multiSigDAOContractId: string;
+}
+
+interface ChangeThresholdForm {
+  threshold: number;
+  safeAccountId: string;
+  multiSigDAOContractId: string;
+}
+
+type UseUpdateMemberParams = (AddMemberForm | DeleteMemberForm | ReplaceMemberForm | ChangeThresholdForm) & {
+  type: MemberOperationsType;
+};
+
+export function useSendProposeTransaction(
```

Comment:
This hook should be decoupled into individual hooks for each mutation action: add owner, remove owner, change owner, change threshold

Reply:
May be we can jump on call to discuss this, so I followed this approach since every member operation is going through to same `proposeTransaction call` with just same `params` plus and minus.